src/components/instructor/AssignmentAdd.js

```diff
@@ -1,19 +1,87 @@
 import React, { useState } from 'react';
+import Dialog from '@mui/material/Dialog';
+import DialogActions from '@mui/material/DialogActions';
+import DialogContent from '@mui/material/DialogContent';
+import DialogTitle from '@mui/material/DialogTitle';
+import Button from '@mui/material/Button';
+import TextField from '@mui/material/TextField';
+import { SERVER_URL } from '../../Constants';
 
-// complete the code.  
-// instructor adds an assignment to a section
-// use mui Dialog with assignment fields Title and DueDate
-// issue a POST using URL /assignments to add the assignment
+const AssignmentAdd = (props) => {
+    const [open, setOpen] = useState(false);
+    const [assignment, setAssignment] = useState({ title: '', dueDate: '' });
+    const [message, setMessage] = useState('');
 
```

Comment:
Add New state to manage saving status
`const [isSaving, setIsSaving] = useState(false);`